File: pag_android/android/src/main/kotlin/dev/hebei/pag_android/PAGViewImpl.kt
package dev.hebei.pag_android

import android.content.Context
import org.libpag.PAGComposition
import org.libpag.PAGView

class PAGViewImpl(registrar: PAGApiPigeonProxyApiRegistrar, private val context: Context) :
    PigeonApiPAGViewApi(registrar) {
    override fun pigeon_defaultConstructor(): PAGView {
        return PAGView(context)
    }

    override fun getComposition(pigeon_instance: PAGView): PAGComposition {
        return pigeon_instance.composition
    }

    override fun setCompositon(pigeon_instance: PAGView, value: PAGComposition) {
        pigeon_instance.composition = value
    }

    override fun getRepeatCount(pigeon_instance: PAGView): Long {
        return pigeon_instance.repeatCount().api
    }

    override fun setRepeatCount(pigeon_instance: PAGView, value: Long) {
        pigeon_instance.setRepeatCount(value.impl)
    }

    override fun getScaleMode(pigeon_instance: PAGView): PAGScaleModeApi {
        return pigeon_instance.scaleMode().scaleModeApi
    }

    override fun setScaleMode(pigeon_instance: PAGView, value: PAGScaleModeApi) {
        pigeon_instance.setScaleMode(value.impl)
    }

    override fun getProgress(pigeon_instance: PAGView): Double {
        return pigeon_instance.progress
    }

    override fun setProgress(pigeon_instance: PAGView, value: Double) {
        pigeon_instance.progress = value
    }

    override fun isPlaying(pigeon_instance: PAGView): Boolean {
        return pigeon_instance.isPlaying
    }

    override fun play(pigeon_instance: PAGView) {
        pigeon_instance.play()
    }

    override fun pause(pigeon_instance: PAGView) {
        pigeon_instance.pause()
    }

    override fun stop(pigeon_instance: PAGView) {
        pigeon_instance.stop()
    }
}

File: pag_darwin/darwin/pag_darwin/Sources/pag_darwin/PAGViewImpl.swift
//
//  PAGViewImpl.swift
//
//  Created by 闫守旺 on 2025/4/9.
//

import libpag

class PAGViewImpl: PigeonApiDelegatePAGViewApi {
    func pigeonDefaultConstructor(pigeonApi: PigeonApiPAGViewApi) throws -> PAGView {
        return PAGView()
    }
    
    func getComposition(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView) throws -> PAGComposition {
        return pigeonInstance.getComposition()
    }
    
    func setCompositon(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView, value: PAGComposition) throws {
        pigeonInstance.setComposition(value)
    }
    
    func getRepeatCount(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView) throws -> Int64 {
        return pigeonInstance.repeatCount().api
    }
    
    func setRepeatCount(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView, value: Int64) throws {
        pigeonInstance.setRepeatCount(value.impl)
    }
    
    func getScaleMode(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView) throws -> PAGScaleModeApi {
        return try pigeonInstance.scaleMode().api
    }
    
    func setScaleMode(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView, value: PAGScaleModeApi) throws {
        pigeonInstance.setScaleMode(value.impl)
    }
    
    func getProgress(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView) throws -> Double {
        return pigeonInstance.getProgress()
    }
    
    func setProgress(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView, value: Double) throws {
        pigeonInstance.setProgress(value)
    }
    
    func isPlaying(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView) throws -> Bool {
        return pigeonInstance.isPlaying()
    }
    
    func play(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView) throws {
        pigeonInstance.play()
    }
    
    func pause(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView) throws {
        pigeonInstance.pause()
    }
    
    func stop(pigeonApi: PigeonApiPAGViewApi, pigeonInstance: PAGView) throws {
        pigeonInstance.stop()
    }
}
